### ISSUE TITLE
Hard to track epoch errors in telemetry

### ISSUE THREAD

**Issue by vladsud** [microsoft/FluidFramework#4515]
It's hard to analyze epoch telemetry data (top-level statistics) as one needs to tack multiple events per session.
Can we instead of having this event

```
            this.logger.sendErrorEvent(
                {
                    eventName: "EpochVersionMismatch",
                    fromCache,
                    clientEpoch: this.fluidEpoch,
                    serverEpoch: epochFromResponse ?? undefined,
                    fetchType,
                },
            error);
```

Add required fields directly to original error object (with error.errorType === OdspErrorType.epochVersionMismatch) ?

Second issue:
There are a lot of fluid:telemetry:OdspDriver:EpochVersionMismatch events with Data_error == "Epoch mispatch" && Data_serverEpoch = "".
409 can (currently) be returned to coherency failure (i.e. not because epoch changes - that's fine, but it shows up with different error message (it will have 409 in its text). error = "Epoch mispatch" is what we generate when epochs are not matching, yet it does not exists in telemetry.  Why?

Query:
> Office_Fluid_FluidRuntime_Error
> | where Data_eventName contains "EpochVersion"
> | where Event_Time > ago(5d)
> | where Data_serverEpoch == ""
> | summarize count() by Data_eventName, Data_error

Results:

	fluid:telemetry:OdspDriver:EpochVersionMismatch	Error 409 from the server, msg = , type = cors	161
	fluid:telemetry:OdspDriver:EpochVersionMismatch	Epoch Mismatch	40
	fluid:telemetry:OdspDriver:EpochVersionMismatch	Error 409 from the server, msg = Conflict, type = cors	17

**Comment by vladsud** [comment]
BTW, having extra properties on actual error object would help to observe and connect (do query joins!) events, like why summarizer errors out with "Epoch mismatch" error.

**Comment by vladsud** [comment]
Here is another anomaly that would be great to get to:
- I see 568 events over last 5 days for fluid:telemetry:OdspDriver:EpochVersionMismatch
- But I see only 391 events for this query:

Office_Fluid_FluidRuntime_Error
| where Data_eventName contains "Epoch" or Data_error contains "epoch" or Data_error contains "409"
| where Data_eventName notcontains "EpochVersion"
| where Event_Time > ago(5d)
| summarize count()

In other words, not all epoch failures show up as failures down stream. I.e. either we have some code that swallows this error and keeps going as nothing happened, or we have code that does not log error while properly handling epoch error.
This error is supposed to be catastrophic, i.e. it needs to result in container being closed, or summarizer to fail (summarizer always issues an error event if it fails).

**Comment by jatgarg** [comment]
For better telemetry: https://github.com/microsoft/FluidFramework/pull/4605

**Comment by jatgarg** [comment]
For differentiating between epoch 409 and coherency 409, this PR was made: https://github.com/microsoft/FluidFramework/pull/4874

**Comment by jatgarg** [comment]
> Here is another anomaly that would be great to get to:
> 
> * I see 568 events over last 5 days for fluid:telemetry:OdspDriver:EpochVersionMismatch
> * But I see only 391 events for this query:
> 
> Office_Fluid_FluidRuntime_Error
> | where Data_eventName contains "Epoch" or Data_error contains "epoch" or Data_error contains "409"
> | where Data_eventName notcontains "EpochVersion"
> | where Event_Time > ago(5d)
> | summarize count()
> 
> In other words, not all epoch failures show up as failures down stream. I.e. either we have some code that swallows this error and keeps going as nothing happened, or we have code that does not log error while properly handling epoch error.
> This error is supposed to be catastrophic, i.e. it needs to result in container being closed, or summarizer to fail (summarizer always issues an error event if it fails).

This stats are now reduced: 4180 vs 4174. With above PR and driver/runtime will be able to differentiate properly between error types and take proper action based on that. However we still need to monitor telemetry as the PR is not in release yet. 

So resolving this issue as it was about telemetry and opening sub issue about anomaly so that after monitoring if it is fixed, then we will be able to resolve this. Otherwise no action to take right now.
